### ISSUE TITLE
Boolean parameters passed via query string are interpreted as strings instead of booleans

### ISSUE THREAD

**Issue by owenconti** [swagger-api/swagger-ui#3302]
Version: 
3.0.17

Swagger file: 
any file will reproduce this issue

Expected:
Passing "true" or "false" as query strings are parsed as booleans into boolean configuration fields.

Result:
"true" or "false" query strings are interpreted as strings, resulting in the following error (in dev):

```
warning.js:36 Warning: Failed prop type: Invalid prop `displayOperationId` of type `string` supplied to `Operation`, expected `boolean`.
    in Operation (created by Operations)
    in Operations (created by _class)
    in _class (created by Connect(_class))
    in Connect(_class) (created by BaseLayout)
    in section (created by Col)
    in Col (created by BaseLayout)
    in div (created by Row)
    in Row (created by BaseLayout)
    in div (created by BaseLayout)
    in div (created by BaseLayout)
    in BaseLayout (created by _class)
    in _class (created by Connect(_class))
    in Connect(_class)
    in section (created by Container)
    in Container
    in StandaloneLayout (created by _class)
    in _class (created by Connect(_class))
    in Connect(_class) (created by App)
    in App (created by _class)
    in _class (created by Connect(_class))
    in Connect(_class) (created by _class2)
    in Provider (created by _class2)
    in _class2
```

URL used to create error: `http://localhost:3200/?displayOperationId=true`

**Comment by shockey** [comment]
I agree with the sentiment here, but note that if we make this change it could break current implementations that rely on `"true"` or `"false"`.

**Comment by webron** [comment]
There aren't too many config options so we can check what makes sense. There's always a problem transferring non-string values as query parameters (like numbers, booleans). We can always say that if a stringified value is valid as well, it needs to be quoted to tell the difference.

**Comment by shockey** [comment]
Another place this bubbles up is with the config defaults: for example, consider `displayOperationId`.

The default is `false: Boolean`. If I were to set it to false explicitly via query parameter, like so:

http://petstore.swagger.io/?displayOperationId=false

... operationIds are displayed, because the value of the option is now `"false": String`. As we know, non-empty strings in JS are truthy, so the option is turned on.

**Comment by shockey** [comment]
Expanding this to include all non-string types that can be expressed in JSON (see also: #3415)

**Comment by shockey** [comment]
Removing `Nice to have`, as this is becoming a pain point.

**Comment by leonardovillela** [comment]
Maybe i can help in this. @shockey u have any solution in mind?
